jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,158 @@
+                                    + "If no dbName is specified, format should be: %s",
+                            params[0], "jdbc:dataproc://hive/;clusterName=<clusterName>"));
+        }
+
+        paramBuilder.setDbName(params.length == 0 ? HIVE_DEFAULT_DATABASE : params[0]);
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        Map<String, String> paramsMap = paramsToMap(params);
+
+        // Get otherSessionConfigs
+        Set<String> nonSessionConfs =
+                ImmutableSet.of(
+                        "projectId",
+                        "region",
+                        "clusterPoolLabel",
+                        "clusterName",
+                        "port",
+                        "httpPath",
+                        "transportMode");
+
+        String otherSessionConfs =
+                paramsMap.entrySet().stream()
+                        .filter(entry -> !nonSessionConfs.contains(entry.getKey()))
+                        .map(Object::toString)
+                        .collect(Collectors.joining(";"));
+        if (!otherSessionConfs.isEmpty()) {
+            paramBuilder.setOtherSessionConfs(otherSessionConfs);
+        }
+
+        // Client shouldn't pass in the session variables that are already set
+        if (paramsMap.containsKey("httpPath") || paramsMap.containsKey("transportMode")) {
+            throw new InvalidURLException(
+                    "Invalid variable.\nPlease do not include httpPath or transportMode.");
+        }
+
+        try {
+            int port =
+                    paramsMap.containsKey("port")
+                            ? Integer.valueOf(paramsMap.get("port"))
+                            : HIVE_DEFAULT_PORT;
+            paramBuilder.setPort(port);
+        } catch (NumberFormatException e) {
+            throw new InvalidURLException(
+                    String.format(
+                            "'port=%s' Please indicate correct port number or remove the field.",
+                            paramsMap.get("port")));
+        }
+
+        String clusterName = paramsMap.getOrDefault("clusterName", null);
+        // TODO: check if cluster name is valid / cluster master node name -m or -m-0
+        if (clusterName == null) {
+            throw new InvalidURLException(
+                    String.format("'clusterName=%s' Please indicate cluster name.", clusterName));
+        }
+        paramBuilder.setHost(String.format("%s-m", clusterName));
+
+        return paramBuilder.build();
+    }
+
+    /**
+     * Helper method that turns the field=value pair into map.
+     *
+     * @param params parameters given by the client
+     * @return Map with fields as keys and has their corresponding values
+     * @throws InvalidURLException
+     */
+    public static Map<String, String> paramsToMap(String[] params) throws InvalidURLException {
```

Comment:
I think we can make it private, or package-private if you have a test case for this method.